src/pages/costModels/costModel/updateMarkupDialog.tsx

```diff
@@ -62,76 +91,111 @@ class UpdateMarkupModelBase extends React.Component<Props, State> {
+              <Text style={styles.cardDescription}>
+                {t('cost_models_details.description_markup_or_discount_model')}
+              </Text>
             </TextContent>
           </StackItem>
           <StackItem>
             <Form>
-              <FormGroup
-                label={t('cost_models_wizard.markup.markup_label')}
-                fieldId="markup-input-box"
-                helperTextInvalid={t('cost_models_wizard.markup.invalid_markup_text')}
-                validated={!isNaN(Number(this.state.markup)) ? 'default' : 'error'}
-              >
-                <InputGroup style={{ width: '150px' }}>
-                  <TextInput
-                    type="text"
-                    aria-label={t('cost_models_wizard.markup.markup_label')}
-                    id="markup-input-box"
-                    value={this.state.markup}
-                    onChange={(markup: string) => {
-                      const markupDecimal = Number(markup);
-                      const dx = markup.split('').findIndex(c => c === '.');
-                      if (!isNaN(markupDecimal) && dx > -1 && markup.length - dx - 1 > 2) {
-                        this.setState({
-                          markup: formatValue(markupDecimal, 'markup', {
-                            fractionDigits: 2,
-                          }) as string,
-                        });
-                        return;
-                      }
-                      this.setState({ markup });
-                    }}
-                    validated={!isNaN(Number(this.state.markup)) ? 'default' : 'error'}
-                  />
-                  <InputGroupText style={{ borderLeft: '0' }}>%</InputGroupText>
-                </InputGroup>
-              </FormGroup>
+              <Grid hasGutter>
+                <GridItem lg={8} id="refSign">
+                  <FormGroup isInline fieldId="markup-or-discount" label={t('cost_models_details.markup_or_discount')}>
+                    <div style={styles.radioAlign}>
+                      <Radio
+                        isChecked={this.state.sign === '+'}
+                        name="markup"
+                        label={t('cost_models_details.markup_plus')}
+                        aria-label={t('cost_models_details.markup_plus')}
+                        id="markup"
+                        value="+"
+                        onChange={this.handleSignChange}
+                      />
+                    </div>
+                    <div style={styles.radioAlign}>
+                      <Radio
+                        isChecked={this.state.sign === '-'}
+                        name="discount"
+                        label={t('cost_models_details.discount_minus')}
+                        aria-label={t('cost_models_details.discount_minus')}
+                        id="discount"
+                        value="-"
+                        onChange={this.handleSignChange}
+                      />
```

Comment:
Instead of maintaining a '-' and "+" value, it would be ideal to just store a boolean variable. For example, store `discount` simply as `true` or `false`.

Reply:
true I can do that and I started out that way, one of the issues there was then I have to constantly convert the sign to a boolean.

Reply:
Maybe I'm misunderstanding, but you don't need to convert the boolean to a string. Unless the user clicks save.

Seems that conversion already happens via the onClick? I was thinking it would look more like:

```
markup: {
  value: this.state.discount ? '-' + this.state.markup : this.state.markup,
  unit: 'percent',
},
```

Reply:
Right I can added another state value, and then use both state.discount and state.sign. I thought you were asking for the sign to be replaced.

Reply:
The other benefit of using boolean is that you don't need to test

isChecked={this.state.sign === '-'} Vs isChecked={this.state.sign === '+'}

If you give the radiobutton the same name (i.e., discount), the radio button would be mutually exclusive. Currently, both buttons could be selected, since they use different names.

Reply:
its here as well
<InputGroupText style={styles.sign}>{this.state.sign}</InputGroupText>

Reply:
I believe this would be preferable to parsing strings in multiple places.

It's also best practice to use a boolean with the radio buttons, ensuring they are mutually exclusive.

Reply:
OK this seems to be working